version: '3'
services:
  # mysql-db:
  #   image: mysql:8.0.29
  #   container_name: mysql-db
  #   ports:
  #     - "3308:3306"
  #   environment:
  #     - MYSQL_DATABASE=practice
  #     - MYSQL_ALLOW_EMPTY_PASSWORD=yes
      
  #   command: --init-file /data/application/init.sql
  #   volumes:
  #     - ./init.sql:/data/application/init.sql
  #     - ./database/data:/var/lib/mysql
  #   healthcheck:
  #       test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
  #       timeout: 3s
  #       retries: 5
  #   networks: 
  #     - mysql-connection
  # db-admin:    
  #   image: phpmyadmin:latest
  #   container_name: mysql-db-admin
  #   restart: always
  #   environment:
  #     PMA_HOST: mysql-db
  #     PMA_USER: root
  #   ports:      
  #     - "8080:80"
  #   depends_on:
  #     mysql-db:
  #       condition: service_healthy
  #   networks: 
  #     - mysql-connection
  # flask_practice:
  #   build: ./flask-practice
  #   container_name: flask_practice
  #   env_file:
  #     - ./flask-practice/.flaskenv
  #   ports:
  #     - 5001:5001
  #   networks: 
  #     - mysql-connection
  #   depends_on:
  #     mysql-db:
  #       condition: service_healthy
  django_practice:
    build: ./django-practice
    container_name: django_practice
    env_file:
      - ./django-practice/.djangoenv
    ports:
      - 5002:5002
    networks: 
      - mysql-connection
    # depends_on:
    #   mysql-db:
    #     condition: service_healthy
networks:
  mysql-connection: